src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -114,6 +115,35 @@ def try_path_munging(
 
 @region_silo_endpoint
 class ProjectStacktraceLinkEndpoint(ProjectEndpoint):  # type: ignore
+    """
+    Returns the index in which the new code mapping should be inserted in the code mapping list.
+    User generated code mappings are evaluated before Sentry generated code mappings.
+    Code mappings with more defined stack trace roots are evaluated before less defined stack trace
+    roots.
+
+    `matched_code_matchings`: The list of code mappings
+    `new_code_mapping`: The new code mapping that is being inserted
+    """
+
+    def get_matched_code_mapping_index(self, matched_code_mappings, new_code_mapping):
+        for index, code_mapping in enumerate(matched_code_mappings):
+            regex = r"\w+"
+            if (
+                code_mapping["automatically_generated"]
+                and not new_code_mapping["automatically_generated"]
+            ) or (
+                code_mapping["automatically_generated"]
+                == new_code_mapping["automatically_generated"]
+                and re.match(
+                    code_mapping["config"]["stackRoot"] + regex,
+                    new_code_mapping["config"]["stackRoot"],
+                )
+            ):
+                return index
+        if new_code_mapping["automatically_generated"]:
+            return len(matched_code_mappings)
```

Comment:
Put code mappings that are Sentry generated and do not have defined stack traces at the end